ext/ddtrace.c

```diff
@@ -1185,6 +1191,71 @@ static PHP_FUNCTION(add_distributed_tag) {
     RETURN_NULL();
 }
 
+static PHP_FUNCTION(set_user) {
+    UNUSED(execute_data);
+
+    zend_string *user_id = NULL;
+    HashTable *metadata = NULL;
+    zend_bool propagate = false;
```

Comment:
I think there should be a config for default usr.id propagation; especially when we will use DDTrace\set_user() in integrations code, that needs to be toggleable anyway. So I propose doing it here.

Reply:
Okay, I'll add a `DD_TRACE_PROPAGATE_USER_ID_DEFAULT` set to false by default in configuration and I'll still keep the parameter so that the user can later still decide regardless of the configuration.

Reply:
Yes, exactly, definitely retain the arg :-)